syntax = "proto3";

message DeviceInfo {
    string device_id = 1;
    string type = 2;      // "camera", "streetlight", etc.
    string ip = 3;
    int32 port = 4;
    string status = 5;    // "ON", "OFF", "ERROR"
}

message SensorData {
    string device_id = 1;
    float value = 2;      // ex: 25.5 (temperatura)
    string timestamp = 3;
}

message Command {
    string device_id = 1;
    string action = 2;    // "TURN_ON", "SET_RESOLUTION", "MULTICAST", etc.
    string parameter = 3; // "1920x1080", "15" (segundos)
}

message Discover {
    string ip = 1;       // ip do gateway
    int32 port_multicast = 2; // porta de resposta para o multicast
    int32 port_udp_sensor = 3; // porta de envio dos dados sensoriais
}

message ListarDispositivos {
    repeated DeviceInfo devices = 1;
}
